File: android/app/src/main/java/com/hindtechgroup/hindpharma/View/DailyCallingView.kt
package com.hindtechgroup.hindpharma.View

import android.content.Intent
import android.net.Uri
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.Button
import androidx.compose.material3.Card
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.RadioButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.hindtechgroup.hindpharma.Models.SessionUser
import com.hindtechgroup.hindpharma.ViewModel.DailyCallingViewModel

@Composable
fun DailyCallingView(user: SessionUser, onBack: () -> Unit, viewModel: DailyCallingViewModel = viewModel()) {
    val context = LocalContext.current
    LaunchedEffect(user.token) { viewModel.load(user.token) }
    Column(Modifier.fillMaxSize().padding(16.dp)) {
        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
            Column { Text(user.businessName ?: "Hind Pharma", style = MaterialTheme.typography.titleLarge); Text("Daily Calling") }
            Button(onClick = onBack) { Text("HOME") }
        }
        viewModel.error?.let { Text(it, color = MaterialTheme.colorScheme.error, modifier = Modifier.padding(vertical = 8.dp)) }
        if (viewModel.loading) CircularProgressIndicator(modifier = Modifier.padding(20.dp))
        LazyColumn(verticalArrangement = Arrangement.spacedBy(10.dp), modifier = Modifier.padding(top = 12.dp)) {
            items(viewModel.items) { medical ->
                Card(Modifier.fillMaxWidth()) {
                    Column(Modifier.padding(14.dp)) {
                        Text(medical.name, style = MaterialTheme.typography.titleMedium)
                        medical.area?.let { Text(it) }
                        Text("Status: ${if (medical.isCall) "Called" else "Not Called"} · ${if (medical.isPick == true) "Picked" else if (medical.isNotPick == true) "Not Picked" else "Pending"}")
                        Button(enabled = !medical.isCall && !viewModel.calling, onClick = {
                            medical.phone?.let { phone ->
                                viewModel.call(medical.id, user.token)
                                context.startActivity(Intent(Intent.ACTION_DIAL, Uri.parse("tel:${phone.replace(Regex("[^0-9+]"), "")}")))
                            }
                        }, modifier = Modifier.fillMaxWidth().padding(top = 8.dp)) { Text(medical.phone ?: "No mobile number") }
                        if (medical.isCall) {
                            Row(Modifier.fillMaxWidth()) {
                                RadioButton(selected = medical.isPick == true, onClick = { viewModel.setStatus(medical.id, true, user.token) })
                                Text("Picked", modifier = Modifier.padding(top = 12.dp))
                                RadioButton(selected = medical.isNotPick == true, onClick = { viewModel.setStatus(medical.id, false, user.token) })
                                Text("Not Picked", modifier = Modifier.padding(top = 12.dp))
                            }
                        }
                    }
                }
            }
        }
    }
}

File: ios/HindPharmaApp/View/DailyCallingView.swift
import SwiftUI

struct DailyCallingView: View {
    let user: SessionUser
    @Environment(\.dismiss) private var dismiss
    @StateObject private var viewModel = DailyCallingViewModel()

    var body: some View {
        List {
            if let error = viewModel.errorMessage { Text(error).foregroundStyle(.red) }
            ForEach(viewModel.items) { medical in
                VStack(alignment: .leading, spacing: 8) {
                    Text(medical.name).font(.headline)
                    if let area = medical.area { Text(area).foregroundStyle(.secondary) }
                    Text("Status: \(medical.isCalled ? "Called" : "Not Called") · \(status(medical))")
                        .font(.subheadline)
                    if let phone = medical.phone, let url = URL(string: "tel://\(phone.filter(\{ $0.isNumber || $0 == "+" \}))") {
                        Button(phone) {
                            Task {
                                await viewModel.call(medical.id, token: user.token)
                                await MainActor.run { UIApplication.shared.open(url) }
                            }
                        }
                        .buttonStyle(.borderedProminent)
                    } else { Text("No mobile number").foregroundStyle(.secondary) }
                    if medical.isCalled {
                        HStack {
                            Button("Picked") { Task { await viewModel.setStatus(medical.id, picked: true, token: user.token) } }
                            Button("Not Picked") { Task { await viewModel.setStatus(medical.id, picked: false, token: user.token) } }
                        }
                    }
                }
                .padding(.vertical, 6)
            }
        }
        .navigationTitle("Daily Calling")
        .overlay { if viewModel.loading { ProgressView() } }
        .task { await viewModel.load(token: user.token) }
    }

    private func status(_ medical: CallingMedical) -> String {
        if medical.is_pick == 1 { return "Picked" }
        if medical.is_not_pick == 1 { return "Not Picked" }
        return "Pending"
    }
}
